File: tests/helpers/plugin-paths.js
'use strict';

const path = require('node:path');
const { artifactRoot: resolveArtifactRoot, validateTarget } = require('../../scripts/artifact-paths');

const repositoryRoot = path.resolve(__dirname, '../..');
const authoringRoot = path.join(repositoryRoot, 'src/harness');
const target = validateTarget(process.env.HARNESS_TEST_TARGET ?? 'development');
const artifactRoot = resolveArtifactRoot(repositoryRoot, target);
const publishedRoot = resolveArtifactRoot(repositoryRoot, 'distribution');

function mapPublishedPath(destination)
{
  const relative = path.relative(publishedRoot, path.resolve(destination));
  if (relative === '..' || relative.startsWith(`..${path.sep}`) || path.isAbsolute(relative))
  {
    return destination;
  }
  return path.join(artifactRoot, relative);
}

module.exports = {
  repositoryRoot,
  authoringRoot,
  target,
  artifactRoot,
  publishedRoot,
  mapPublishedPath,
  authoringPath: (...segments) => path.join(authoringRoot, ...segments),
  artifactPath: (...segments) => path.join(artifactRoot, ...segments),
};

File: tests/write-asd-ste100/plugin_paths.py
"""Select the same installation-shaped test artifact as the Node helpers."""
import os
from pathlib import Path

REPOSITORY_ROOT = Path(__file__).resolve().parents[2]
TARGET = os.environ.get('HARNESS_TEST_TARGET', 'development')
TARGETS = {'development': '.build/harness', 'distribution': 'dist/harness'}
if TARGET not in TARGETS:
    raise ValueError(f'Invalid HARNESS_TEST_TARGET: {TARGET!r}')
ARTIFACT_ROOT = REPOSITORY_ROOT / TARGETS[TARGET]
SCRIPTS = ARTIFACT_ROOT / 'skills/write-asd-ste100/scripts'
